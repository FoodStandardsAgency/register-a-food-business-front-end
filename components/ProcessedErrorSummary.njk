{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}

{% macro processedErrorSummary(validatorErrors, language) %}
  {% if validatorErrors | length %}
    {{ govukErrorSummary({
        titleText: __({ phrase: "There is a problem", locale: language }),
        errorList: validatorErrors | selectValidationErrors(language)
    }) }}
  {% endif %}
{% endmacro %}