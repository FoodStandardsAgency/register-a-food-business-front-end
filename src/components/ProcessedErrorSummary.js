import { ErrorSummary } from "govuk-react";
import PropTypes from "prop-types";
import { withTranslation } from "../../i18n.js";

const ProcessedErrorSummary = (props) => {
  const errors = [];
  for (let error in props.validatorErrors) {
    errors.push({
      targetName: error,
      text: props.t(props.validatorErrors[error])
    });
  }
  if (Object.keys(props.validatorErrors).length > 0) {
    return (
      <ErrorSummary
        id="errorSummary"
        heading={props.t("There is a problem")}
        onHandleErrorClick={props.onHandleErrorClick}
        errors={errors}
        role="alert"
      />
    );
  } else {
    return null;
  }
};
export default withTranslation("common")(ProcessedErrorSummary);

ProcessedErrorSummary.propTypes = {
  validatorErrors: PropTypes.objectOf(PropTypes.string)
};
