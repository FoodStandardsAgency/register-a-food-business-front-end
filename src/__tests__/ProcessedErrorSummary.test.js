import ProcessedErrorSummary from "../components/ProcessedErrorSummary";
import { shallow, mount } from "enzyme";

const testValidatorErrorsExists = {
  example: "test error",
  example2: "test error 2"
};

const emptyValidatorErrors = {};

describe("<ProcessedErrorSummary />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(
      <ProcessedErrorSummary validatorErrors={testValidatorErrorsExists} />
    );
    expect(wrapper.length).toBe(1);
  });

  it("returns null when validator errors is empty", () => {
    const wrapper = mount(
      <ProcessedErrorSummary validatorErrors={emptyValidatorErrors} />
    );
    expect(wrapper.instance()).toEqual(null);
  });

  it("should click on the error", () => {
    const mockOnHandleErrorClickCallback = jest.fn();

    const ProcessedErrorSummaryClick = () => (
      <ProcessedErrorSummary
        onHandleErrorClick={mockOnHandleErrorClickCallback}
        validatorErrors={testValidatorErrorsExists}
      />
    );

    const wrapperProcessedErrorSummaryClickMock = mount(
      <ProcessedErrorSummaryClick />
    );

    let timesClicked = 0;

    wrapperProcessedErrorSummaryClickMock
      .find("UnorderedList")
      .find("ListItem")
      .forEach(listItem => {
        listItem.find("Anchor").simulate("click");
        timesClicked += 1;
        expect(mockOnHandleErrorClickCallback.mock.calls.length).toBe(
          timesClicked
        );
      });
  });
});
