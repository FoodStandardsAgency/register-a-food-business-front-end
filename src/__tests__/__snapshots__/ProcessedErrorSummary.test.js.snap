// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProcessedErrorSummary /> matches the previous snapshot 1`] = `
.emotion-6 {
  color: #0b0c0c;
  padding: 15px;
  border: 4px solid #b10e1e;
  margin-bottom: 20px;
}

.emotion-6:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0px;
}

@media only screen and (min-width:641px) {
  .emotion-6 {
    padding: 20px;
    border: 5px solid #b10e1e;
  }
}

@media only screen and (min-width:641px) {
  .emotion-6 {
    margin-bottom: 30px;
  }
}

.emotion-0 {
  font-family: "nta",Arial,sans-serif;
  font-weight: bold;
  margin: 0;
  font-size: 24px;
  line-height: 1.0416666666666667;
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 36px;
    line-height: 1.1111111111111112;
    margin-bottom: 20px;
  }
}

@media print {
  .emotion-0 {
    font-size: 18px;
  }
}

.emotion-5 {
  margin: 0;
  padding: 0;
  padding-left: 20px;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  margin-bottom: 0;
}

@media only screen and (min-width:641px) {
  .emotion-5 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .emotion-5 {
    margin-bottom: 0;
  }
}

.emotion-2 {
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
  width: 100%;
  margin-bottom: 0;
}

@media only screen and (min-width:641px) {
  .emotion-2 {
    font-size: 16px;
    line-height: 1.25;
    margin-bottom: 5px;
  }
}

@media only screen and (min-width:641px) {
  .emotion-2 {
    margin-bottom: 0;
  }
}

.emotion-1 {
  color: #005ea5;
  padding: 3px;
  margin: -3px;
  outline-color: transparent;
  display: inline-block;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  margin-bottom: 5px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-skip-ink: none;
  text-decoration-skip-ink: none;
  text-transform: none;
  cursor: pointer;
  font-size: 16px;
  line-height: 1.25;
  color: #b10e1e;
  padding-top: 4px;
  padding-bottom: 2px;
}

.emotion-1:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

.emotion-1:visited {
  color: #2e358b;
}

.emotion-1:visited {
  color: #b10e1e;
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    font-size: 19px;
    line-height: 1.3;
  }
}

<div
  className="emotion-6"
  tabIndex={-1}
>
  <h2
    className="emotion-0"
  >
    There is a problem
  </h2>
  <ul
    className="emotion-5"
  >
    <li
      className="emotion-2"
    >
      <a
        className="emotion-1"
        onClick={[Function]}
        tabIndex={1}
      >
        test error
      </a>
    </li>
    <li
      className="emotion-2"
    >
      <a
        className="emotion-1"
        onClick={[Function]}
        tabIndex={2}
      >
        test error 2
      </a>
    </li>
  </ul>
</div>
`;
